src/logging/log_manager.cpp

```diff
@@ -0,0 +1,155 @@
+	log_storage = make_uniq<InMemoryLogStorage>(db);
+}
+
+LogManager::~LogManager() {
+}
+
+void LogManager::Initialize() {
+	LoggingContext context(LogContextScope::DATABASE);
+	global_logger = CreateLogger(context, true, true);
+}
+
+LogManager &LogManager::Get(ClientContext &context) {
+	return context.db->GetLogManager();
+}
+
+RegisteredLoggingContext LogManager::RegisterLoggingContextInternal(LoggingContext &context) {
+	RegisteredLoggingContext result = {next_registered_logging_context_index, context};
+
+	next_registered_logging_context_index += 1;
+
+	if (next_registered_logging_context_index == NumericLimits<idx_t>::Maximum()) {
+		throw InternalException("Ran out of available log context ids.");
+	}
+
+	return result;
+}
+
+void LogManager::WriteLogEntry(timestamp_t timestamp, const char *log_type, LogLevel log_level, const char *log_message,
+                               const RegisteredLoggingContext &context) {
+	unique_lock<mutex> lck(lock);
+	log_storage->WriteLogEntry(timestamp, log_level, log_type, log_message, context);
+}
+
+void LogManager::FlushCachedLogEntries(DataChunk &chunk, const RegisteredLoggingContext &context) {
+	throw NotImplementedException("FlushCachedLogEntries");
+}
+
+void LogManager::SetEnableLogging(bool enable) {
+	unique_lock<mutex> lck(lock);
+	config.enabled = enable;
+	global_logger->UpdateConfig(config);
+}
+
+void LogManager::SetLogMode(LogMode mode) {
+	unique_lock<mutex> lck(lock);
+	config.mode = mode;
+	global_logger->UpdateConfig(config);
+}
+
+void LogManager::SetLogLevel(LogLevel level) {
+	unique_lock<mutex> lck(lock);
+	config.level = level;
+	global_logger->UpdateConfig(config);
+}
+
+void LogManager::SetEnabledLogTypes(unordered_set<string> &enabled_log_types) {
+	unique_lock<mutex> lck(lock);
+	config.enabled_log_types = enabled_log_types;
+	global_logger->UpdateConfig(config);
+}
+
+void LogManager::SetDisabledLogTypes(unordered_set<string> &disabled_log_types) {
+	unique_lock<mutex> lck(lock);
+	config.disabled_log_types = disabled_log_types;
+	global_logger->UpdateConfig(config);
+}
+
+void LogManager::SetLogStorage(DatabaseInstance &db, const string &storage_name) {
+	unique_lock<mutex> lck(lock);
+	auto storage_name_to_lower = StringUtil::Lower(storage_name);
+
+	if (config.storage == storage_name_to_lower) {
+		return;
+	}
+
+	// Flush the old storage, we are going to replace it.
+	log_storage->Flush();
+
+	if (storage_name_to_lower == LogConfig::IN_MEMORY_STORAGE_NAME) {
+		log_storage = make_shared_ptr<InMemoryLogStorage>(db);
+	} else if (storage_name_to_lower == LogConfig::STDOUT_STORAGE_NAME) {
+		log_storage = make_shared_ptr<StdOutLogStorage>();
+	} else if (storage_name_to_lower == LogConfig::FILE_STORAGE_NAME) {
+		throw NotImplementedException("File log storage is not yet implemented");
+	} else if (registered_log_storages.find(storage_name_to_lower) != registered_log_storages.end()) {
+		log_storage = registered_log_storages[storage_name_to_lower];
+	}
```

Comment:
@samansmink: should there be a:
```
else {
    throw NotImplementedException("Log Storage %s not implemented and not registered", storage_name_to_lower);
}
```

Currently:
```
SET logging_storage='abc';
```
succeeds but do not actually change `log_storage`.